lib/fibonacci.rb

```diff
@@ -1,8 +1,39 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - 0(n)
+# 0(n * 3) Since I have 3 variables, but if we get rid of the constant. 
+# Will be O(n)
 # Hint, you may want a recursive helper method
 def fibonacci(n)
  
 end
+
+class Fibonacci
+  def self.fibonacci(n)
```

Comment:
This works, but you really don't need to make a class with class variables.  It makes more sense as a method.